src/react/backend/location/LocationDetails/__tests__/LocationDetailsAzureArchive.test.tsx

```diff
@@ -0,0 +1,428 @@
+  );
+};
+
+type prefilledValues = [RegExp, string][];
+const prefilledHelper = (container: HTMLElement, values: prefilledValues) => {
+  values.forEach((v) => {
+    userEvent.type(getByLabelText(container, v[0]), v[1]);
+  });
+  values.forEach((v) => {
+    expect(getByLabelText(container, v[0])).toHaveValue(v[1]);
+  });
+};
+
+const serviceBusTopicHelper = (container: HTMLElement) => {
+  const preDefinedAzureServiceBusTopic: [RegExp, string][] = [
+    [/Topic Name \*/, 'mock-topic-name'],
+    [/Topic Subscription ID \*/, 'mock-subscription-id'],
+    [/Namespace/, 'mock-namespace'],
+  ];
+  prefilledHelper(container, preDefinedAzureServiceBusTopic);
+};
+
+const serviceBusQueueHelper = (container: HTMLElement) => {
+  const preDefinedAzureServiceBusTopic: [RegExp, string][] = [
+    [/Queue Name \*/, 'mock-queue-name'],
+    [/Queue Endpoint \*/, 'mock-endpoint'],
+  ];
+  prefilledHelper(container, preDefinedAzureServiceBusTopic);
+};
+
+const storageQueueHelper = (container: HTMLElement) => {
+  const preDefinedAzureServiceBusTopic: [RegExp, string][] = [
+    [/Queue Name \*/, 'mock-queue-name'],
+    [/Queue Endpoint/, 'mock-endpoint'],
+  ];
+  prefilledHelper(container, preDefinedAzureServiceBusTopic);
+};
+
+const authClientSecretHelper = (container: HTMLElement) => {
+  const values: [RegExp, string][] = [
+    [/Tenant ID \*/, 'mock-tenant-id'],
+    [/Azure Client ID \*/, 'mock-azure-client-id'],
+    [/Azure Client Key \*/, 'mock-azure-client-key'],
+  ];
+  prefilledHelper(container, values);
+};
+
+const authSharedAccessSignature = (
+  container: HTMLElement,
+  isServiceBus = false,
+) => {
+  const values: [RegExp, string][] = [
+    [/SAS token for Storage \*/, 'mock-sas-token-for-storage'],
+  ];
+
+  if (isServiceBus) {
+    values.push([
+      /SAS token for ServiceBus * \*/,
+      'mock-sas-token-for-service-bus',
+    ]);
+  }
+  prefilledHelper(container, values);
+};
+
+const authSharedKey = (container: HTMLElement) => {
+  const values: [RegExp, string][] = [
+    [/Azure Account Name \*/, 'mock-azure-account-name'],
+    [/Azure Account Key \*/, 'mock-azure-account-key'],
+  ];
+  prefilledHelper(container, values);
+};
+
+describe('<LocationDetailsAzureArchive />', () => {
+  it('should show azure details for empty details', () => {
+    //S
+    setupAndRenderLocationDetails();
+
+    //V
+    const textGrep = [
+      'Endpoint *',
+      /target azure container name \*/i,
+      /Queue type \*/i,
+      /Topic Name \*/i,
+      /topic subscription id \*/i,
+      /namespace/i,
+      //   /Authentication type \*/i,
+      /Tenant ID \*/i,
+      /Azure Client ID \*/i,
+      /Azure Client Key \*/i,
+    ];
+
+    // Test more stuff
+
+    textGrep.forEach((text) => {
+      expect(screen.getByLabelText(text)).toBeInTheDocument();
+      expect(screen.getByLabelText(text)).toHaveValue('');
+    });
+  });
+
+  // usefull for later
+
+  test('Azure Service Bus Topic + Client Secret', async () => {
```

Comment:
We should also check the `Azure Shared Key` is disable for the `Azure Service Bus Topic`.